core/core/services/phones.py

```diff
@@ -0,0 +1,34 @@
+import re
+from core.models import response, Record
+from core.database import write_data, Database
+from core.misc import InfoMessages
+
+
+database = Database()
+
+
+@response(InfoMessages.PHONE_NUMBER_ADDED)
+@write_data
+def add_phone_number(payload):
+    record = database[payload.name]
+
+    return set_phone_number(payload, record)
+
+
+def set_phone_number(payload, record: Record) -> Record:
+    if record:
+        if payload.phone not in record.phones:
+            record.phones.add(payload.phone)
+        return record
+    else:
+        return Record(name=payload.name, phones=set([payload.phone]))
```

Comment:
set already store only unique values, so we don't need to check if `payload.phone not in record.phones` and also if record exist than it is should contain empty set by default in `phones` attribute, so we don't need to check if phones exist too.
it should work like this:
```
def set_phone_number(payload, record: Record) -> Record:
    if record:
        record.phones.add(payload.phone)
        return record
    else:
        return Record(name=payload.name, phones={payload.phone})

Reply:
oh...yeah../that's a set, not a list. Got it😊